### ISSUE TITLE
Panic loading some wikipedia content

### ISSUE THREAD

**Issue by fabricedesre** [servo/servo#30703]
Tested on commit 1db357456a . I'm sure this used to work but did not get a chance to bisect.

Steps to reproduce:
- Launch `./servo https://en.wikipedia.org/wiki/Freediving`

Expected:
- A nice wikipedia article.

Observed:
- A very unexpected panic:
```
index out of bounds: the len is 2 but the index is 2 (thread Style#0, at /home/capyloon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/xi-unicode-0.1.0/src/lib.rs:43)
[2023-11-06T22:18:47Z ERROR servo::main2] index out of bounds: the len is 2 but the index is 2
Layout thread disconnected: "SendError(..)" (thread Script(1,1), at components/script/dom/window.rs:1912)
[2023-11-06T22:18:47Z ERROR servo::main2] Layout thread disconnected: "SendError(..)"
called `Result::unwrap()` on an `Err` value: PoisonError { .. } (thread Layout(1,2), at components/layout_thread_2020/lib.rs:962)
[2023-11-06T22:18:47Z ERROR servo::main2] called `Result::unwrap()` on an `Err` value: PoisonError { .. }
[2023-11-06T22:18:47Z ERROR constellation::constellation] crash page crashed
condition was Some(DirtyDescendants) (thread Script(1,2), at components/script/dom/window.rs:2001)
[2023-11-06T22:18:47Z ERROR servo::main2] condition was Some(DirtyDescendants)
[2023-11-06T22:18:47Z ERROR constellation::constellation] crash page crashed

```

**Comment by sagudev** [comment]
Did you used nightly builds from servo.org or did you build it yourself. Release builds from mach now have debug assertions enabled by default, while nightly builds do not. You can try production profile to disable them.

**Comment by fabricedesre** [comment]
I get the same crash with a production profile and with the release one (all local builds).

**Comment by sagudev** [comment]
Do we have any tool for bisect of servo nightly builds?

**Comment by jdm** [comment]
There's https://github.com/servo/servo/pull/19947 (and subsequent windows and macis support), but you still need to do the bisection by hand. https://github.com/servo/servo/issues/19505 was never completed.

**Comment by atouchet** [comment]
I was able to bisect this and the first nightly build where this page started crashing was 2023-07-21.

Regression range: https://github.com/servo/servo/compare/59b0b1bade0add818f57d20e80f9322b5a0834e3...2b67392fd509393cc0c66ab4292e48b04926ab5e

**Comment by mrego** [comment]
On the bisect topic, if we ever got flatpak support, then [flatpak-bisect](https://github.com/flatpak/flatpak/blob/main/scripts/flatpak-bisect) could be useful for that.

**Comment by mrobinson** [comment]
> On the bisect topic, if we ever got flatpak support, then [flatpak-bisect](https://github.com/flatpak/flatpak/blob/main/scripts/flatpak-bisect) could be useful for that.

All nightlies are uploaded to S3 so it should be possible to bisect from there in a platform-independent way.

**Comment by mrobinson** [comment]
Here's a reduced test case for this issue:

```html
<br><img>&nbsp;
```

**Comment by sagudev** [comment]
> On the bisect topic, if we ever got flatpak support, then [flatpak-bisect](https://github.com/flatpak/flatpak/blob/main/scripts/flatpak-bisect) could be useful for that.

We would need bisect in `mach` to support non-linux platforms too.

**Comment by fabricedesre** [comment]
Re-opening since with a build from f1c291853e331329271efba52a03ba5049e8358b I still get:

```shell
$ ./servo https://en.wikipedia.org/wiki/Freediving
index out of bounds: the len is 2 but the index is 2 (thread Style#0, at /home/capyloon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/xi-unicode-0.1.0/src/lib.rs:43)
[2023-11-30T17:22:38Z ERROR servo::main2] index out of bounds: the len is 2 but the index is 2
Layout thread disconnected: "SendError(..)" (thread Script(1,1), at components/script/dom/window.rs:1912)
[2023-11-30T17:22:39Z ERROR servo::main2] Layout thread disconnected: "SendError(..)"
called `Result::unwrap()` on an `Err` value: PoisonError { .. } (thread Layout(1,2), at components/layout_thread_2020/lib.rs:962)
[2023-11-30T17:22:39Z ERROR servo::main2] called `Result::unwrap()` on an `Err` value: PoisonError { .. }
[2023-11-30T17:22:39Z ERROR constellation::constellation] crash page crashed
condition was Some(DirtyDescendants) (thread Script(1,2), at components/script/dom/window.rs:2001)
[2023-11-30T17:22:39Z ERROR servo::main2] condition was Some(DirtyDescendants)
[2023-11-30T17:22:39Z ERROR constellation::constellation] crash page crashed
```

**Comment by fabricedesre** [comment]
> It looks like this is a build that compiles against a commit of servo older than the fix in [f1c2918](https://github.com/servo/servo/commit/f1c291853e331329271efba52a03ba5049e8358b).

Indeed, I was syncing up from an old "production" build instead of the current "release" one. Sorry for the noise.
